### PR TITLE
fix set_cleanliness_flags to work with toggle

### PR DESCRIPTION
**Pull request by czue**

### CODE REVIEW

**Comment by czue** [comment]
@snopoke cc @emord

File: corehq/ex-submodules/casexml/apps/phone/cleanliness.py [hunk 1]
```diff
@@ -30,9 +31,21 @@ def set_cleanliness_flags(domain, owner_id):
...
+            cleanliness_obj.is_clean and not OWNERSHIP_CLEANLINESS.enabled(domain)
+        ) or (
+            # if dirty, first check the hint and only do a full check if it's not valid
+            not cleanliness_object.is_clean and (
+                not cleanliness_object.hint or not hint_still_valid(domain, owner_id, cleanliness_object.hint)
+            )
+        )
     if not cleanliness_object.is_clean:
-        if not cleanliness_object.hint or not hint_still_valid(domain, owner_id, cleanliness_object.hint):
+        if needs_full_check(domain, cleanliness_object):
```
Comment (snopoke) [inline corehq/ex-submodules/casexml/apps/phone/cleanliness.py]: Should this still be inside the parent `if not cleanliness_object.is_clean`

Comment (czue) [inline corehq/ex-submodules/casexml/apps/phone/cleanliness.py]: nope. thanks for catching. https://github.com/dimagi/commcare-hq/commit/759b7746c7e7f5812490be5a81176ad91dfaec46

**Comment by snopoke** [comment]
:+1: merge on tests passing

**Pull request merged**